alica_test_utility/include/alica/test/SuccessBehaviour.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "IMockUpBehaviour.h"
+
+#include <cstdint>
+
+namespace alica::test
+{
+template <uint32_t iterationsBeforeSuccess>
+class SuccessBehaviour : public IMockUpBehaviour
+{
+public:
+    SuccessBehaviour();
+
+    void run(void* msg) override;
+
+private:
+    uint32_t _iterationsBeforeSuccess;
+};
+
+template <uint32_t iterationsBeforeSuccess>
+SuccessBehaviour<iterationsBeforeSuccess>::SuccessBehaviour()
+        : IMockUpBehaviour("SuccessBehaviour")
+        , _iterationsBeforeSuccess(iterationsBeforeSuccess)
+{
+}
+
+template <uint32_t iterationsBeforeSuccess>
+void SuccessBehaviour<iterationsBeforeSuccess>::run(void* msg)
+{
+    if (iterationsCounter() >= _iterationsBeforeSuccess && !this->isSuccess()) {
+        this->setSuccess();
```

Comment:
"this" is almost always implied and can be skipped

Reply:
done